### ISSUE TITLE
VB.NET AnalyzeDataFlow incorrectly reports parameters as WrittenInside

### ISSUE THREAD

**Issue by TymurGubayev** [dotnet/roslyn#68286]
**Version Used**: 
4.6.0

**Steps to Reproduce**:

Use following source code to analyse
```vb
Public Sub Test(i As Integer)
    Dim s = i.ToString()
End Sub
```
with following visitor
```cs
public override SyntaxNode VisitMethodBlock(MethodBlockSyntax node)
{
    var first = node.Statements.First();
    var last = node.Statements.Last();
    var dataFlow = semanticModel.AnalyzeDataFlow(first, last);
    Debug.Assert(dataFlow.WrittenInside.Length == 1);

    return node;
}
```

**Expected Behavior**:
`dataFlow.WrittenInside` should not contain `Parameter i As System.Int32`, only `Local s As System.String`

**Actual Behavior**:
`dataFlow.WrittenInside` contains `Parameter i As System.Int32`.

**Notes**:
Equivalent code in C# is analysed as expected, the parameter is not part of the `dataFlow.WrittenInside`.
